### PR TITLE
Conditionally make calls to PerformanceObserver.takeRecords which doesnt exist in Safari

### PR DESCRIPTION
**Pull request by ernieturner**
Number of other Prettier changes in this changeset in files that I assume haven't been saved since the config was introduced.

### CODE REVIEW

File: src/observe.ts [hunk 1]
```diff
@@ -33,7 +34,9 @@ export const disconnectPerfObserversHidden = (): void => {
     poDisconnect(2);
   }
   if (perfObservers[3]) {
-    perfObservers[3].takeRecords();
+    if (typeof perfObservers[3].takeRecords === 'function') {
+      perfObservers[3].takeRecords();
+    }
```
Comment (ernieturner) [inline src/observe.ts:39]: I assumed that the calls to `logMetric` and `poDisconnect` should still happen in this scenario, but let me know if that isn't the case.

Comment (Zizzamia) [reply src/observe.ts:39]: Interesting point, let me think out loud the logic.
Actually more I re-think the logic out loud more I think there is one more bug to solve 🤦🏻‍♂️
The final records are probably never saved in `cls.value`.

File: __tests__/firstInput.spec.ts [hunk 2]
```diff
@@ -46,5 +46,13 @@ describe('firstInput', () => {
       expect(spy).toHaveBeenCalledWith(2, 'cls');
       expect(spy).toHaveBeenCalledWith(3, 'tbt');
     });
+
+    it('should conditionally call takeRecords', () => {
```
Comment (Zizzamia) [inline __tests__/firstInput.spec.ts:50]: Thank you for adding the test

**Comment by Zizzamia** [comment]
@ernieturner just published [v5.2.0](https://github.com/Zizzamia/perfume.js/releases/tag/v5.2.0) with your fix. Thank you again 🙏

**Pull request merged**